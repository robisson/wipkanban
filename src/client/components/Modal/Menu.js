import React from 'react';

const Menu = () => {
    return (
        <div>
            <a id="addChecklist" href="#" className="btn btn-sm btn-primary">
                <span className="fa fa-check-square"></span>
                Checklist
            </a>
            <span id="containerDataTarefa">
                <span>
                    <a href="#" id="btnMostrarDatepicker" className="hide btn btn-sm btn-primary">
                        <span className="fa fa-calendar"></span>
                        Definir data
                    </a>
                    <span id="datePickerDataTarefa" className="hide">
                        <input
                            size="8"
                            maxLength="10"
                            type="text"
                            defaultValue="14/08/2017"
                            className="datepicker text-center"/>
                        <button id="btnDefinirData" className="btn btn-success btn-sm">OK</button>
                    </span>
                    <span className="" id="dataSelecionada">
                        <span className="badge badge-success">
                            <input type="checkbox"/>
                            <span title="Clique para alterar a data" id="dataTarefa">14/08/2017</span>
                        </span>
                        <button id="btnRemover" className="btn btn-sm btn-danger" title="Remover data">
                            <i className="fa fa-remove"></i>
                        </button>
                    </span>
                </span>
            </span>
            <a id="btnMarcarTempo" href="#" className="btn btn-sm btn-primary">
                <span className="fa fa-clock-o"></span>
                Marcar tempo
            </a>
            <a href="#" className="btn btn-sm btn-danger" id="excluirTarefa">
                <span className="fa fa-remove"></span>
                Excluir
            </a>
        </div>
    );
}

export default Menu;