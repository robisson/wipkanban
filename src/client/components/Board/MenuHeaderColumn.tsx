import * as React from "react";
import { withStyles, StyleRulesCallback } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import CallSplitIcon from "@material-ui/icons/CallSplit";
import ListItemText from "@material-ui/core/ListItemText";
import MoreVertIcon from "@material-ui/icons/MoreVert";

const ITEM_HEIGHT = 48;

const styles: StyleRulesCallback = () => ({
  menuItem: {
    paddingTop: 2,
    paddingBottom: 2
  }
});

type State = {
  anchorEl?: any | null;
};

type Props = {
  classes: any;
};

class MenuHeaderColumn extends React.Component<Props, State> {
  state: State = {
    anchorEl: null
  };

  handleClick = (event: React.SyntheticEvent) => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { anchorEl } = this.state;
    const { classes } = this.props;

    return (
      <div>
        <IconButton
          style={{
            color: "#ffffff"
          }}
          aria-label="More"
          aria-owns={anchorEl ? "long-menu" : ""}
          aria-haspopup="true"
          onClick={this.handleClick}
        >
          <MoreVertIcon />
        </IconButton>
        <Menu
          id="long-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
          PaperProps={{
            style: {
              maxHeight: ITEM_HEIGHT * 4.5,
              width: 200
            }
          }}
        >
          <MenuItem onClick={this.handleClose} className={classes.menuItem}>
            <ListItemIcon>
              <CallSplitIcon />
            </ListItemIcon>
            <ListItemText inset primary="Vertical Split" />
          </MenuItem>
          <MenuItem onClick={this.handleClose} className={classes.menuItem}>
            <ListItemIcon>
              <CallSplitIcon />
            </ListItemIcon>
            <ListItemText inset primary="Horizontal Split" />
          </MenuItem>
        </Menu>
      </div>
    );
  }
}

export default withStyles(styles)(MenuHeaderColumn);
