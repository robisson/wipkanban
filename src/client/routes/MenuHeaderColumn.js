// @flow
import * as React from "react";
import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import MoreVertIcon from '@material-ui/icons/MoreVert';

const ITEM_HEIGHT = 48;

type State = {
  anchorEl?: Object
}

class MenuHeaderColumn extends React.Component<State> {
  state : State = {
    anchorEl: null
  };

  handleClick = (event) => {
    this.setState({anchorEl: event.currentTarget});
  };

  handleClose = () => {
    this.setState({anchorEl: null});
  };

  render() {
    const {anchorEl} = this.state;

    return (
      <div>
        <IconButton
          style={{
          color: '#ffffff'
        }}
          aria-label="More"
          aria-owns={anchorEl
          ? 'long-menu'
          : null}
          aria-haspopup="true"
          onClick={this.handleClick}>
          <MoreVertIcon/>
        </IconButton>
        <Menu
          id="long-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
          PaperProps={{
          style: {
            maxHeight: ITEM_HEIGHT * 4.5,
            width: 200
          }
        }}>
          <MenuItem onClick={this.handleClose}>Profile</MenuItem>
          <MenuItem onClick={this.handleClose}>My account</MenuItem>
          <MenuItem onClick={this.handleClose}>Logout</MenuItem>
        </Menu>
      </div>
    );
  }
}

export default MenuHeaderColumn;
