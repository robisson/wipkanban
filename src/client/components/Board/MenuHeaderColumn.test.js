import React from "react";
import MenuHeaderColumn from "./MenuHeaderColumn";
import { createShallow } from "@material-ui/core/test-utils";
import toJson from "enzyme-to-json";
import { withStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";

// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("<MenuHeaderColumn /> Component", () => {
  let shallow;
  let wrapper;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    wrapper = shallow(<MenuHeaderColumn />);
  });

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render 1 div with Item Button and Menu component", () => {
    expect(wrapper.type()).toEqual("div");
    expect(wrapper.find(IconButton).type()).toEqual(IconButton);
    expect(wrapper.find(Menu).type()).toEqual(Menu);
    expect(wrapper.find(MenuItem).length).toEqual(2);
  });
});
