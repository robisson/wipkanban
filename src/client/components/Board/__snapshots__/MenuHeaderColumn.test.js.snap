// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuHeaderColumn /> Component should render correctly 1`] = `
<div>
  <WithStyles(IconButton)
    aria-haspopup="true"
    aria-label="More"
    aria-owns={null}
    onClick={[Function]}
    style={
      Object {
        "color": "#ffffff",
      }
    }
  >
    <pure(MoreVertIcon) />
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    PaperProps={
      Object {
        "style": Object {
          "maxHeight": 216,
          "width": 200,
        },
      }
    }
    anchorEl={null}
    id="long-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      className="MenuHeaderColumn-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Vertical Split"
      />
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      className="MenuHeaderColumn-menuItem-1"
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <pure(CallSplitIcon) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        inset={true}
        primary="Horizontal Split"
      />
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</div>
`;
